packages/builder/src/stores/builder/components.js

```diff
@@ -803,7 +816,10 @@ export class ComponentStore extends BudiStore {
         // sibling
         const previousSibling = parent._children[index - 1]
         const definition = this.getDefinition(previousSibling._component)
-        if (definition.hasChildren) {
+        if (
+          definition.hasChildren &&
+          componentTreeNodes[`nodeOpen-${previousSibling._id}`]
```

Comment:
Could we add a function to the component tree node store to check if a component is visible or not? 
i.e. `componentTreeNodesStore.isNodeExpanded(previousSibling._id)`

That would be a lot nicer than having to use this syntax with the magic string `nodeOpen` all over the place. Think it would be a good idea to hide the implementation details away from the usages.

Reply:
Good idea, done 👍